src/components/Select/MultiSelect.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import ChevronDownIcon from '@moda/icons/chevron-down-12';
+import ChevronUpIcon from '@moda/icons/chevron-up-12';
+import ExitIcon from '@moda/icons/exit-16';
+import { SelectableOption } from '.';
+import { SelectOptions } from './SelectOptions';
+import { useSelect } from './useSelect';
+import './MultiSelect.scss';
+
+type Props = Omit<React.HTMLAttributes<HTMLDivElement>, 'defaultValue' | 'value' | 'onChange'> & {
+  defaultValue?: string[] | undefined;
+  disabled?: boolean;
+  idRef?: string;
+  name?: string;
+  placeholder?: string;
+  onChange?: (value: string[]) => void;
+  onRemoveSelectedItem: (value: string) => void;
+  options: SelectableOption[];
+  value?: string[] | undefined;
+  error?: boolean | string;
+  shiftIconLeftwards?: boolean;
+  ignoreCasing?: boolean;
+  dropDirection?: 'down' | 'up';
+};
+
+export const MultiSelect: React.FC<Props> = ({
+  idRef = '',
+  className,
+  options,
+  name,
+  disabled,
+  value,
+  placeholder,
+  onRemoveSelectedItem,
+  onChange,
+  defaultValue,
+  error,
+  shiftIconLeftwards = false,
+  ignoreCasing = false,
+  dropDirection = 'down',
+  ...rest
+}) => {
+  const { state, dispatch, Mode, selectRef } = useSelect({ value, defaultValue });
+  const [filteredValue, setFilteredValue] = useState('');
+
+  const handleSelect = useCallback(
+    (option: SelectableOption) => {
+      dispatch({ type: 'SELECT', payload: { value: option.value } });
+      setFilteredValue('');
+      onChange && onChange(value ? [...value, option.value] : [option.value]);
```

Comment:
I think this is a bit confusing.
`dispatch({ type: 'SELECT', payload: { value: option.value } });` sets `value` in the state, but we're never using that value from the state.

So I'd make this consistent with `Select` and use `state.value` as the source of truth. That way the component will be uncontrolled (i.e. you don't have to specify `value` and `onChange` for it to _work_).